auth/mtls/mtls.go

```diff
@@ -55,6 +58,70 @@ type CredentialsLoader interface {
 	// LoadServerCertificate returns the certificate that should be presented
 	// by the server to incoming clients.
 	LoadServerCertificate(context.Context) (tls.Certificate, error)
+
+	// CertRefreshed indicates if internally any of the cert data has
+	// been refreshed and should be reloaded. This will depend on the
+	// implementation to support but allows for dynamic refresh of certificates
+	// without a server restart.
+	CertsRefreshed() bool
+}
+
+type WrappedTransportCredentials struct {
+	creds      credentials.TransportCredentials
+	ctx        context.Context
```

Comment:
can this field be removed?

Reply:
Yeah, was torn here. Technically to generate new credentials we need a context so it was either cache one or use context.Background(). I removed it and used context.Background instead which feels a little cleaner since cached ones might be expired.

Mainly because the credentials interface doesn't pass creds through each function, only some.